### PR TITLE
fix(consume): convert bounds to selected quantity unit

### PR DESCRIPTION
**Pull request by lujoga**
The minimum/maximum value of the amount field for products with tare weight handling is broken on the `/consume` page. The bounds should be multiplied by the conversion factor of the selected quantity unit, otherwise the validation will complain about out-of-bounds input.

A screenshot of the upper half of the consume form showing the described issue

The `/purchase` page seems to just circumvent this by forcing the selected unit to be the stock unit, but I'm not sure if this is the right solution.

### CODE REVIEW

**Comment by berrnd** [comment]
Please be so kind and provide a practical example what this is about.

So what shouldn't work right now and the rational behind the changes.

**Review by berrnd — CHANGES_REQUESTED** [review]
Best guess what this should refer to based on a live persistent demo example: https://test-tod320235armkgip1l1tjo.demo-prerelease.grocy.info/consume

- Take the product "Example1" on the just mentioned demo instance.
- It's tracked in Packs and each Pack contains 100 Pieces.
- Let's assume 1 Pack is in stock.
- => It's already perfectly possible to consume either 1 Pack or 100 Pieces (or less) and the amount picker's `max` attribute is therefore of course already handled accordingly:

https://github.com/grocy/grocy/assets/2161815/7c94c2a0-287e-49c5-a4e1-0bd62354bc58

So please explain a little bit what this should be a about. Having me guessing all that by providing such a pull request is frankly annoying and it wastes my time.

File: public/viewjs/consume.js [hunk 1]
```diff
@@ -476,13 +476,13 @@ Grocy.Components.ProductPicker.GetPicker().on('change', function(e)
 
 				if (productDetails.product.enable_tare_weight_handling == 1)
 				{
-					$("#display_amount").attr("min", productDetails.product.tare_weight);
-					$('#display_amount').attr('max', productDetails.stock_amount + productDetails.product.tare_weight);
+					$("#display_amount").attr("min", productDetails.product.tare_weight * $("#qu_id option:selected").attr("data-qu-factor"));
+					$("#display_amount").attr("max", (productDetails.stock_amount + productDetails.product.tare_weight) * $("#qu_id option:selected").attr("data-qu-factor"));
```
[the remark this answers is not in the document]
Comment (lujoga) [reply public/viewjs/consume.js:480]: In this case, we need to multiply the tare weight by the conversion factor as well, since the amount given by the user contains it, but that may be specified in a quantity unit other than the stock QU.

File: public/viewjs/consume.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿$('#save-consume-button').on('click', function(e)
+$('#save-consume-button').on('click', function(e)
```
Comment (lujoga) [inline public/viewjs/consume.js:1]: This is just removing the UTF-8 BOM since my editor kept doing that.

File: public/viewjs/consume.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (berrnd) [reply public/viewjs/consume.js:480]: As of today, by definition the product's tare weight is related to the product's QU stock (and that's exactly outlined like so on the product edit page). So if you want to make the tare weight "selected QU aware", feel free to do so, but then by far this here is not the only place to take care of.

**Comment by berrnd** [comment]
Thanks for finally providing a description.

&nbsp;
> The `/purchase` page seems to just circumvent this by forcing the selected unit to be the stock unit, but I'm not sure if this is the right solution.

As of today, by definition the product's tare weight is related to the product's QU stock (and that's exactly outlined like so on the product edit page). That means it only works when using the product's QU stock on all related places (/consume, /purchase, everywhere) and that's a known limitation of the current tare weight handling design.

If you want to remove that limitation, this here is by far not the only thing which needs to be extended to make this possible.

Please test stuff practically thoroughly, keeping also edge cases in mind, and provide a easy understandable description for any further pull request. Providing one like this one doesn't help, it wastes (my) time and only creates frustration on the reviewer's (again, my) end.

File: public/viewjs/consume.js (hunk 2 above)
Comment (berrnd) [reply public/viewjs/consume.js:1]: I haven't blamed that, I've blamed changing single quotes to double quotes for no reason (not in that line this particular comment was pinned on).
